packages/s3-store/index.ts

```diff
@@ -741,4 +809,26 @@ export class S3Store extends DataStore {
 
     return deleted
   }
+
+  private async uniqueTmpFileName(template: string): Promise<string> {
+    const fileName = template + crypto.randomBytes(10).toString('base64url').slice(0, 10)
+    const filePath = path.join(os.tmpdir(), fileName)
+
+    const fileExists = await new Promise<boolean>((resolve, reject) => {
+      fs.lstat(filePath, (error) => {
+        if (!error) {
+          return resolve(true)
+        }
+        if (error.code === 'ENOENT') {
+          return resolve(false)
+        }
+        reject(error)
+      })
+    })
```

Comment:
Would be nice to start to transition to `fs/promises` where possible, especially new additions.

Reply:
Yes, I just found it less verbose using the callback instead of `try { } catch {} ` for this instance.
But happy to migrate to fs/promises if preferred

Reply:
Done